### DISCUSSION TITLE
Multiple search with fixed selections

### DISCUSSION THREAD

**Discussion by smartm0use** [fomantic/Fomantic-UI#1858] [Modules (Modal, Dropdown, Popup, Calendar, Toast, ...)] [answered]
Hello, is it possibile to have a multiple selection dropdown that have one or more fixed seletions that user can't remove?
I mean something like this:

![image](https://user-images.githubusercontent.com/5399265/105980177-d468db00-6094-11eb-970e-c9c7065a5fb6.png)

In this case Alaska is a default value that can't be removed by user.

**Comment by lubber-de** [comment, marked as answer]
Two options here:

- Use the callback `onLabelRemove` and return false when a desired label is tried to be removed
- Use a custom label renderer template which in addition does not render the remove icon on desired labels

See an example using both options: The preselected label "english" cannot be removed the usual way anymore
https://jsfiddle.net/lubber/bxc2n5yp/12/

**Comment by smartm0use** [comment]
Hello lubber-de, thank you for your great hint!

I noticed that `label` in `onLabelRemove` contains `value` instead of displayed `text` of the selected item.
Since my check is on the `text`, can I use any workaround to get it from the removed item?

**Reply by ko2in** [reply]
What's the difference to check with text instead of value?

**Reply by lubber-de** [reply]
> Since my check is on the `text`, can I use any workaround to get it from the removed item?

Sure. Just extract the Text from the label itself (which is available as `this` context
```javascript
return this.text() !== 'English'
```
See https://jsfiddle.net/lubber/bxc2n5yp/19/

**Comment by smartm0use** [comment]
That the value could be "AL" and the text "Alabama".

**Reply by ko2in** [reply]
Yes. I know that, but I can't see any difference by checking text or value. Whatever you check, they belong the same item.

**Comment by smartm0use** [comment]
Hello, what in case of multiple fixed selections? I have to deal with an array of fixed items and I have to match with selected ones...

**Reply by lubber-de** [reply]
Use an array instead then. See https://jsfiddle.net/lubber/bxc2n5yp/24/

**Reply by smartm0use** [reply]
Many thanks!
